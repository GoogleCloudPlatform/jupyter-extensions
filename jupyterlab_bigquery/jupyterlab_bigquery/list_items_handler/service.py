# Lint as: python3
"""Request handler classes for the extensions."""

import re
from google.cloud import bigquery
from google.cloud.datacatalog import DataCatalogClient, enums, types

SCOPE = ("https://www.googleapis.com/auth/cloud-platform",)


class BigQueryService:
  """Provides an authenticated BigQuery Client and project info"""

  _instance = None

  def __init__(self,
               client=bigquery.Client(),
               datacatalog_client=DataCatalogClient()):
    self._client = client
    self._datacatalog_client = datacatalog_client

  @property
  def client(self):
    return self._client

  @classmethod
  def get(cls):
    if not cls._instance:
      cls._instance = BigQueryService()
    return cls._instance

  def list_projects(self):
    project = self._client.project
    projects_list = {
      format(project): {
        'id': format(project),
        'name': format(project),
      }
    }
    project_ids = [format(project)]
    return {'projects': projects_list, 'projectIds': project_ids}

  def list_datasets(self, project):
    datasets = list(self._client.list_datasets(project))

    datasets_list = {}
    dataset_ids = []
    for dataset in datasets:
      dataset_full_id = '{}.{}'.format(dataset.project, dataset.dataset_id)
      datasets_list[dataset_full_id] = {
          'id': dataset_full_id,
          'name': dataset.dataset_id,
          'projectId': format(dataset.project),
      }
      dataset_ids.append(dataset_full_id)

    return {'datasets': datasets_list, 'datasetIds': dataset_ids}

  def list_tables(self, dataset_id):
    dataset = self._client.get_dataset(dataset_id)
    tables = list(self._client.list_tables(dataset))

    tables_list = {}
    table_ids = []
    for table in tables:
      table_full_id = '{}.{}.{}'.format(table.project, table.dataset_id, table.table_id)
      tables_list[table_full_id] = {
        'id': table_full_id,
        'name': table.table_id,
        'datasetId': dataset_id,
        'type': table.table_type,
        'partitioned': True if table.partitioning_type else False,
      }
      table_ids.append(table_full_id)

    return {'tables': tables_list, 'tableIds': table_ids}

  def list_models(self, dataset_id):
    dataset = self._client.get_dataset(dataset_id)
    models = list(self._client.list_models(dataset))

    models_list = {}
    model_ids = []
    for model in models:
      model_full_id = '{}.{}.{}'.format(model.project, model.dataset_id, model.model_id)
      models_list[model_full_id] = {
        'id': model_full_id,
        'name': model.model_id,
        'datasetId': dataset_id,
      }
      model_ids.append(model_full_id)

    return {'models': models_list, 'modelIds': model_ids}

  def search_projects(self, search_key, project_id):
    scope = types.SearchCatalogRequest.Scope()
    scope.include_project_ids.append(project_id)
    results = self._datacatalog_client.search_catalog(
        scope=scope,
        query='name:{} projectid:{}'.format(search_key, project_id))

    fetched_results = []
    for result in results:
      resource = result.linked_resource
      result_type = format(result.search_result_subtype)
      if result_type == 'entry.dataset':
        res = re.search('datasets/(.*)', resource)
        dataset = res.group(1)
        fetched_results.append({
            'type': 'dataset',
            'parent': project_id,
            'name': dataset,
            'id': '{}.{}'.format(project_id, dataset)
        })
      elif result_type == 'entry.table':
        res = re.search('datasets/(.*)/tables/(.*)', resource)
        dataset = res.group(1)
        table = res.group(2)
        fetched_results.append({
            'type': 'table',
            'parent': dataset,
            'name': table,
            'id': '{}.{}.{}'.format(project_id, dataset, table)
        })
      elif result_type == 'entry.table.view':
        res = re.search('datasets/(.*)/tables/(.*)', resource)
        dataset = res.group(1)
        view = res.group(2)
        fetched_results.append({
            'type': 'view',
            'parent': dataset,
            'name': view,
            'id': '{}.{}.{}'.format(project_id, dataset, view)
        })
      elif result_type == 'entry.model':
        res = re.search('datasets/(.*)/models/(.*)', resource)
        dataset = res.group(1)
        model = res.group(2)
        fetched_results.append({
            'type': 'model',
            'parent': dataset,
            'name': model,
            'id': '{}.{}.{}'.format(project_id, dataset, model)
        })

    return {'results': fetched_results}

  def create_custom_client(self, project_id):
    return bigquery.Client(project=project_id)

  def get_project(self, custom_client):
    project = custom_client.project
    formatted_project = {
        'id': format(project),
        'name': format(project),
    }
    return formatted_project
