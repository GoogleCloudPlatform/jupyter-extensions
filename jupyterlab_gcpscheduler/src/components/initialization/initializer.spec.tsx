/**
 * Copyright 2020 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { shallow } from 'enzyme';
import { CheckboxInput, SubmitButton } from 'gcp_jupyterlab_shared';
import * as React from 'react';

import { ProjectState, ProjectStateService } from '../../service/project_state';
import { getProjectState } from '../../test_helpers';
import { Initializer } from './initializer';

describe('Initializer', () => {
  const mockGetProjectState = jest.fn();
  const mockCreateBucket = jest.fn();
  const mockEnableServices = jest.fn();
  const mockCreateCloudFunction = jest.fn();
  const mockDialogClose = jest.fn();
  const mockOnInitialized = jest.fn();
  const mockProjectStateService = ({
    createBucket: mockCreateBucket,
    createCloudFunction: mockCreateCloudFunction,
    enableServices: mockEnableServices,
    getProjectState: mockGetProjectState,
  } as undefined) as ProjectStateService;
  let mockProjectState: ProjectState;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProjectState = getProjectState();
  });

  it('Renders with validating message', () => {
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );
    expect(initializer).toMatchSnapshot();
  });

  it('Renders with error if project state cannot be determined', async () => {
    const getStatePromise = Promise.reject('No project found');
    mockGetProjectState.mockReturnValue(getStatePromise);
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    expect(initializer).toMatchSnapshot();
  });

  it('Renders with ready message', async () => {
    mockProjectState.schedulerRegion = 'us-central1';
    mockProjectState.canSubmitImmediateJobs = true;
    mockProjectState.canSubmitScheduledJobs = true;
    const getStatePromise = Promise.resolve(mockProjectState);
    mockGetProjectState.mockReturnValue(getStatePromise);
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    expect(initializer).toMatchSnapshot();
    expect(mockOnInitialized).toHaveBeenCalled();
  });

  it('Renders with app engine content', async () => {
    mockProjectState.allServicesEnabled = true;
    mockProjectState.hasGcsBucket = true;
    mockProjectState.hasCloudFunction = true;
    const getStatePromise = Promise.resolve(mockProjectState);
    mockGetProjectState.mockReturnValue(getStatePromise);
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    expect(initializer).toMatchSnapshot();

    // Simulate Check Again
    initializer.find(SubmitButton).simulate('click');
    expect(initializer).toMatchSnapshot('Rechecking');
    await getStatePromise.catch(() => null);
    expect(initializer).toMatchSnapshot('After recheck');
    expect(mockGetProjectState).toHaveBeenCalledTimes(2);
  });

  it('Renders with initialization content', async () => {
    const getStatePromise = Promise.resolve(mockProjectState);
    mockGetProjectState.mockReturnValue(getStatePromise);
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    expect(initializer).toMatchSnapshot();
  });

  it('Renders with initialization content for immediate runs only', async () => {
    const getStatePromise = Promise.resolve(mockProjectState);
    mockGetProjectState.mockReturnValue(getStatePromise);
    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    initializer
      .find(CheckboxInput)
      .simulate('change', { target: { checked: false } });
    expect(initializer).toMatchSnapshot();
  });

  it('Runs initialization sequence', async () => {
    const getStatePromise = Promise.resolve(mockProjectState);
    const enabledServicesPromise = Promise.resolve(true);
    const createBucketPromise = Promise.resolve(true);
    const deployFunctionPromise = Promise.resolve(true);

    mockGetProjectState.mockReturnValueOnce(getStatePromise);
    mockEnableServices.mockReturnValue(enabledServicesPromise);
    mockCreateBucket.mockReturnValue(createBucketPromise);
    mockCreateCloudFunction.mockReturnValue(deployFunctionPromise);

    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    initializer.find(SubmitButton).simulate('click');
    expect(initializer.state('isActivatingServices')).toBe(true);
    await enabledServicesPromise.catch(() => null);
    expect(initializer.state('isActivatingServices')).toBe(false);
    expect(initializer.state('isCreatingBucket')).toBe(true);
    expect(initializer.state('isDeployingFunction')).toBe(true);

    await createBucketPromise.catch(() => null);
    await deployFunctionPromise.catch(() => null);
    expect(initializer.state('isCreatingBucket')).toBe(false);
    expect(initializer.state('isDeployingFunction')).toBe(false);
    expect(initializer).toMatchSnapshot('Shows AppEngine creator');

    expect(mockEnableServices).toHaveBeenCalledWith([
      'storage-api.googleapis.com',
      'ml.googleapis.com',
      'cloudscheduler.googleapis.com',
      'cloudfunctions.googleapis.com',
    ]);
    expect(mockCreateBucket).toHaveBeenCalled();
    expect(mockCreateCloudFunction).toHaveBeenCalled();
    // Gets called after all steps complete
    expect(mockGetProjectState).toHaveBeenCalledTimes(2);
  });

  it('Fails to enable services', async () => {
    const getStatePromise = Promise.resolve(mockProjectState);
    const enabledServicesPromise = Promise.reject('Could not enable services');
    const createBucketPromise = Promise.resolve(true);
    const deployFunctionPromise = Promise.resolve(true);

    mockGetProjectState.mockReturnValueOnce(getStatePromise);
    mockEnableServices.mockReturnValue(enabledServicesPromise);
    mockCreateBucket.mockReturnValue(createBucketPromise);
    mockCreateCloudFunction.mockReturnValue(deployFunctionPromise);

    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    initializer.find(SubmitButton).simulate('click');
    expect(initializer.state('isActivatingServices')).toBe(true);
    await enabledServicesPromise.catch(() => null);
    expect(initializer).toMatchSnapshot();
    expect(initializer.state('serviceActivationError')).toBe(
      'Could not enable services: Unable to enable necessary GCP APIs'
    );
    expect(initializer.state('isActivatingServices')).toBe(false);
    expect(initializer.state('isCreatingBucket')).toBe(false);
    expect(initializer.state('isDeployingFunction')).toBe(false);

    expect(mockEnableServices).toHaveBeenCalledWith([
      'storage-api.googleapis.com',
      'ml.googleapis.com',
      'cloudscheduler.googleapis.com',
      'cloudfunctions.googleapis.com',
    ]);
    expect(mockCreateBucket).not.toHaveBeenCalled();
    expect(mockCreateCloudFunction).not.toHaveBeenCalled();
  });

  it('Fails to create resource', async () => {
    const getStatePromise = Promise.resolve(mockProjectState);
    const enabledServicesPromise = Promise.resolve(true);
    const createBucketPromise = Promise.reject('Could not create bucket');
    const deployFunctionPromise = Promise.resolve(true);

    mockGetProjectState.mockReturnValueOnce(getStatePromise);
    mockEnableServices.mockReturnValue(enabledServicesPromise);
    mockCreateBucket.mockReturnValue(createBucketPromise);
    mockCreateCloudFunction.mockReturnValue(deployFunctionPromise);

    const initializer = shallow(
      <Initializer
        projectStateService={mockProjectStateService}
        onDialogClose={mockDialogClose}
        onInitialized={mockOnInitialized}
      />
    );

    await getStatePromise.catch(() => null);
    initializer.find(SubmitButton).simulate('click');
    expect(initializer.state('isActivatingServices')).toBe(true);
    await enabledServicesPromise.catch(() => null);
    expect(initializer.state('isActivatingServices')).toBe(false);
    expect(initializer.state('isCreatingBucket')).toBe(true);
    expect(initializer.state('isDeployingFunction')).toBe(true);
    await createBucketPromise.catch(() => null);
    await deployFunctionPromise.catch(() => null);
    expect(initializer).toMatchSnapshot();
    expect(initializer.state('bucketCreationError')).toBe(
      'Could not create bucket: Unable to create Cloud Storage bucket'
    );
    expect(initializer.state('isCreatingBucket')).toBe(false);
    expect(initializer.state('isDeployingFunction')).toBe(false);

    expect(mockEnableServices).toHaveBeenCalledWith([
      'storage-api.googleapis.com',
      'ml.googleapis.com',
      'cloudscheduler.googleapis.com',
      'cloudfunctions.googleapis.com',
    ]);
    expect(mockCreateBucket).toHaveBeenCalled();
    expect(mockCreateCloudFunction).toHaveBeenCalled();
    expect(mockGetProjectState).toHaveBeenCalledTimes(1);
  });
});
