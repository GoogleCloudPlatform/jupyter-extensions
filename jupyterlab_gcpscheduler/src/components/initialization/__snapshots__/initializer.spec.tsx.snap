// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initializer Fails to create resource 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <CheckboxInput
      checked={true}
      disabled={false}
      label="Enable Recurring Scheduled Runs?"
      name="enableRecurring"
      onChange={[Function]}
      title="If checked, jobs can be created to run on a recurring schedule"
    />
    <ServiceStatuses
      isActivating={false}
      serviceStatuses={
        Array [
          Object {
            "enabled": true,
            "service": Object {
              "documentation": "https://cloud.google.com/storage/",
              "endpoint": "storage-api.googleapis.com",
              "isOptional": false,
              "name": "Cloud Storage API",
            },
          },
          Object {
            "enabled": true,
            "service": Object {
              "documentation": "https://cloud.google.com/ai-platform/",
              "endpoint": "ml.googleapis.com",
              "isOptional": false,
              "name": "AI Platform Training API",
            },
          },
          Object {
            "enabled": true,
            "service": Object {
              "documentation": "https://cloud.google.com/scheduler",
              "endpoint": "cloudscheduler.googleapis.com",
              "isOptional": true,
              "name": "Cloud Scheduler API",
            },
          },
          Object {
            "enabled": true,
            "service": Object {
              "documentation": "https://cloud.google.com/cloud-build/",
              "endpoint": "cloudbuild.googleapis.com",
              "isOptional": true,
              "name": "Cloud Build API",
            },
          },
          Object {
            "enabled": true,
            "service": Object {
              "documentation": "https://cloud.google.com/functions/",
              "endpoint": "cloudfunctions.googleapis.com",
              "isOptional": true,
              "name": "Cloud Functions API",
            },
          },
        ]
      }
    />
    <ResourceStatuses
      bucketCreationError="Could not create bucket: Unable to create Cloud Storage bucket"
      isCreatingBucket={false}
      isDeployingFunction={false}
      projectState={
        Object {
          "allServicesEnabled": true,
          "canSubmitImmediateJobs": false,
          "canSubmitScheduledJobs": false,
          "hasCloudFunction": true,
          "hasGcsBucket": false,
          "projectId": "test-project",
          "requiredServicesEnabled": false,
          "schedulerRegion": "",
          "serviceStatuses": Array [
            Object {
              "enabled": true,
              "service": Object {
                "documentation": "https://cloud.google.com/storage/",
                "endpoint": "storage-api.googleapis.com",
                "isOptional": false,
                "name": "Cloud Storage API",
              },
            },
            Object {
              "enabled": true,
              "service": Object {
                "documentation": "https://cloud.google.com/ai-platform/",
                "endpoint": "ml.googleapis.com",
                "isOptional": false,
                "name": "AI Platform Training API",
              },
            },
            Object {
              "enabled": true,
              "service": Object {
                "documentation": "https://cloud.google.com/scheduler",
                "endpoint": "cloudscheduler.googleapis.com",
                "isOptional": true,
                "name": "Cloud Scheduler API",
              },
            },
            Object {
              "enabled": true,
              "service": Object {
                "documentation": "https://cloud.google.com/cloud-build/",
                "endpoint": "cloudbuild.googleapis.com",
                "isOptional": true,
                "name": "Cloud Build API",
              },
            },
            Object {
              "enabled": true,
              "service": Object {
                "documentation": "https://cloud.google.com/functions/",
                "endpoint": "cloudfunctions.googleapis.com",
                "isOptional": true,
                "name": "Cloud Functions API",
              },
            },
          ],
        }
      }
      showCloudFunction={true}
    />
    <p>
      Click the Initialize button to finish configuring
       test-project
      .
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Check Again"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Fails to enable services 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <CheckboxInput
      checked={true}
      disabled={false}
      label="Enable Recurring Scheduled Runs?"
      name="enableRecurring"
      onChange={[Function]}
      title="If checked, jobs can be created to run on a recurring schedule"
    />
    <ServiceStatuses
      activationError="Could not enable services: Unable to enable necessary GCP APIs"
      isActivating={false}
      serviceStatuses={
        Array [
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/storage/",
              "endpoint": "storage-api.googleapis.com",
              "isOptional": false,
              "name": "Cloud Storage API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/ai-platform/",
              "endpoint": "ml.googleapis.com",
              "isOptional": false,
              "name": "AI Platform Training API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/scheduler",
              "endpoint": "cloudscheduler.googleapis.com",
              "isOptional": true,
              "name": "Cloud Scheduler API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/cloud-build/",
              "endpoint": "cloudbuild.googleapis.com",
              "isOptional": true,
              "name": "Cloud Build API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/functions/",
              "endpoint": "cloudfunctions.googleapis.com",
              "isOptional": true,
              "name": "Cloud Functions API",
            },
          },
        ]
      }
    />
    <ResourceStatuses
      isCreatingBucket={false}
      isDeployingFunction={false}
      projectState={
        Object {
          "allServicesEnabled": false,
          "canSubmitImmediateJobs": false,
          "canSubmitScheduledJobs": false,
          "hasCloudFunction": false,
          "hasGcsBucket": false,
          "projectId": "test-project",
          "requiredServicesEnabled": false,
          "schedulerRegion": "",
          "serviceStatuses": Array [
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/storage/",
                "endpoint": "storage-api.googleapis.com",
                "isOptional": false,
                "name": "Cloud Storage API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/ai-platform/",
                "endpoint": "ml.googleapis.com",
                "isOptional": false,
                "name": "AI Platform Training API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/scheduler",
                "endpoint": "cloudscheduler.googleapis.com",
                "isOptional": true,
                "name": "Cloud Scheduler API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/cloud-build/",
                "endpoint": "cloudbuild.googleapis.com",
                "isOptional": true,
                "name": "Cloud Build API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/functions/",
                "endpoint": "cloudfunctions.googleapis.com",
                "isOptional": true,
                "name": "Cloud Functions API",
              },
            },
          ],
        }
      }
      showCloudFunction={true}
    />
    <p>
      Click the Initialize button to finish configuring
       test-project
      .
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Check Again"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with app engine content 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <AppEngineCreator
      projectId="test-project"
    />
    <p>
      Click the Check Again button after you have completed the steps listed above.
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Check Again"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with app engine content: After recheck 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <AppEngineCreator
      projectId="test-project"
    />
    <p>
      Click the Check Again button after you have completed the steps listed above.
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Check Again"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with app engine content: Rechecking 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <Message
    asActivity={true}
    asError={false}
    text="Validating project configuration..."
  />
</div>
`;

exports[`Initializer Renders with error if project state cannot be determined 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <Message
    asActivity={false}
    asError={true}
    text="No project found: Unable to determine project status"
  />
</div>
`;

exports[`Initializer Renders with initialization content 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <CheckboxInput
      checked={true}
      disabled={false}
      label="Enable Recurring Scheduled Runs?"
      name="enableRecurring"
      onChange={[Function]}
      title="If checked, jobs can be created to run on a recurring schedule"
    />
    <ServiceStatuses
      isActivating={false}
      serviceStatuses={
        Array [
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/storage/",
              "endpoint": "storage-api.googleapis.com",
              "isOptional": false,
              "name": "Cloud Storage API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/ai-platform/",
              "endpoint": "ml.googleapis.com",
              "isOptional": false,
              "name": "AI Platform Training API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/scheduler",
              "endpoint": "cloudscheduler.googleapis.com",
              "isOptional": true,
              "name": "Cloud Scheduler API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/cloud-build/",
              "endpoint": "cloudbuild.googleapis.com",
              "isOptional": true,
              "name": "Cloud Build API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/functions/",
              "endpoint": "cloudfunctions.googleapis.com",
              "isOptional": true,
              "name": "Cloud Functions API",
            },
          },
        ]
      }
    />
    <ResourceStatuses
      isCreatingBucket={false}
      isDeployingFunction={false}
      projectState={
        Object {
          "allServicesEnabled": false,
          "canSubmitImmediateJobs": false,
          "canSubmitScheduledJobs": false,
          "hasCloudFunction": false,
          "hasGcsBucket": false,
          "projectId": "test-project",
          "requiredServicesEnabled": false,
          "schedulerRegion": "",
          "serviceStatuses": Array [
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/storage/",
                "endpoint": "storage-api.googleapis.com",
                "isOptional": false,
                "name": "Cloud Storage API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/ai-platform/",
                "endpoint": "ml.googleapis.com",
                "isOptional": false,
                "name": "AI Platform Training API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/scheduler",
                "endpoint": "cloudscheduler.googleapis.com",
                "isOptional": true,
                "name": "Cloud Scheduler API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/cloud-build/",
                "endpoint": "cloudbuild.googleapis.com",
                "isOptional": true,
                "name": "Cloud Build API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/functions/",
                "endpoint": "cloudfunctions.googleapis.com",
                "isOptional": true,
                "name": "Cloud Functions API",
              },
            },
          ],
        }
      }
      showCloudFunction={true}
    />
    <p>
      Click the Initialize button to finish configuring
       test-project
      .
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Initialize"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with initialization content for immediate runs only 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <CheckboxInput
      checked={false}
      disabled={false}
      label="Enable Recurring Scheduled Runs?"
      name="enableRecurring"
      onChange={[Function]}
      title="If checked, jobs can be created to run on a recurring schedule"
    />
    <ServiceStatuses
      isActivating={false}
      serviceStatuses={
        Array [
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/storage/",
              "endpoint": "storage-api.googleapis.com",
              "isOptional": false,
              "name": "Cloud Storage API",
            },
          },
          Object {
            "enabled": false,
            "service": Object {
              "documentation": "https://cloud.google.com/ai-platform/",
              "endpoint": "ml.googleapis.com",
              "isOptional": false,
              "name": "AI Platform Training API",
            },
          },
        ]
      }
    />
    <ResourceStatuses
      isCreatingBucket={false}
      isDeployingFunction={false}
      projectState={
        Object {
          "allServicesEnabled": false,
          "canSubmitImmediateJobs": false,
          "canSubmitScheduledJobs": false,
          "hasCloudFunction": false,
          "hasGcsBucket": false,
          "projectId": "test-project",
          "requiredServicesEnabled": false,
          "schedulerRegion": "",
          "serviceStatuses": Array [
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/storage/",
                "endpoint": "storage-api.googleapis.com",
                "isOptional": false,
                "name": "Cloud Storage API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/ai-platform/",
                "endpoint": "ml.googleapis.com",
                "isOptional": false,
                "name": "AI Platform Training API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/scheduler",
                "endpoint": "cloudscheduler.googleapis.com",
                "isOptional": true,
                "name": "Cloud Scheduler API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/cloud-build/",
                "endpoint": "cloudbuild.googleapis.com",
                "isOptional": true,
                "name": "Cloud Build API",
              },
            },
            Object {
              "enabled": false,
              "service": Object {
                "documentation": "https://cloud.google.com/functions/",
                "endpoint": "cloudfunctions.googleapis.com",
                "isOptional": true,
                "name": "Cloud Functions API",
              },
            },
          ],
        }
      }
      showCloudFunction={false}
    />
    <p>
      Click the Initialize button to finish configuring
       test-project
      .
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Initialize"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with ready message 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <Message
      asActivity={false}
      asError={false}
      text="Project test-project is ready"
    />
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="OK"
      />
    </ActionBar>
  </div>
</div>
`;

exports[`Initializer Renders with validating message 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <Message
    asActivity={true}
    asError={false}
    text="Validating project configuration..."
  />
</div>
`;

exports[`Initializer Runs initialization sequence: Shows AppEngine creator 1`] = `
<div
  className="column_fghohlb"
>
  <p>
    Run a notebook from start to finish immediately or on a recurring schedule. The executed notebook output will be saved in a Cloud Storage bucket and viewable from a dashboard. Using this feature will incur additional charges. By clicking Initialize, you are agreeing to the terms of service for the APIs and services listed below.
  </p>
  <div>
    <AppEngineCreator
      projectId="test-project"
    />
    <p>
      Click the Check Again button after you have completed the steps listed above.
    </p>
    <ActionBar
      onDialogClose={[MockFunction]}
    >
      <SubmitButton
        actionPending={false}
        onClick={[Function]}
        text="Check Again"
      />
    </ActionBar>
  </div>
</div>
`;
